tests/model/test_cycle_data.py

```diff
@@ -300,8 +304,15 @@ def test_performance_signatures_are_deleted(test_perf_signature):
 
 
 def test_try_data_removal(
-    try_repository, test_repository, try_push_stored, test_perf_signature, test_perf_signature_2
+    try_repository,
+    test_repository,
+    try_push_stored,
+    test_perf_signature,
+    test_perf_signature_2,
+    monkeypatch,
 ):
+    monkeypatch.setattr(taskcluster, 'Notify', MagicMock())
```

Comment:
The need to monkey patch this whenever we cycle Perfherder data feels brittle.
It's easy for a future developer to not know about this (or forget) when adding a new Perfherder cycling test. Then the email recipients can easily be spammed while the tests are running on CI.

We need to look for a safer solution.